lib/Runtime/Debug/DiagHelperMethodWrapper.cpp

```diff
@@ -212,8 +212,7 @@ namespace Js
         // Note: there also could be plain OutOfMemoryException and StackOverflowException, no special handling for these.
         if (!exceptionObject->IsDebuggerSkip() ||
             exceptionObject == scriptContext->GetThreadContext()->GetPendingOOMErrorObject() ||
-            exceptionObject == scriptContext->GetThreadContext()->GetPendingSOErrorObject() ||
-            !scriptContext)
```

Comment:
The assert above won't cover this in release builds. I think it's worth adding `!scriptContext ||` above line 214.